File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set =new HashSet<>();
        int st=0,end=0;

        // for(int i=0;i<s.length;i++){
        //     st.add(s.chatAt(i));
        // }
        int maxlen=0;
        while(end<s.length()){
            char c=s.charAt(end);

            while(set.contains(c)){
                set.remove(s.charAt(st));
                st++;
            }
            set.add(c);
            maxlen=Math.max(maxlen,end-st+1);
            end++;
        }
        return maxlen;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxlen=0;
        // int hash[255]={0};
        vector<int> hash(255,0);
        for(int i=0;i<s.size();i++){
        //    fill(hash,hash+255,0);
        fill(hash.begin(),hash.end(),0);
            int len=0;
            for(int j=i;j<s.size();j++){
                if(hash[s[j]]==1) break;
                len=j-i+1;
                maxlen=max(maxlen,len);
                hash[s[j]]=1;
            }
        }
        return maxlen;
    }
};